txpool/txpool_test.go

```diff
@@ -98,60 +93,79 @@ type result struct {
 /* Single account cases (unit tests) */
 
 func TestAddTxErrors(t *testing.T) {
+	poolSigner := crypto.NewEIP155Signer(100)
+
+	// Generate a private key and address
+	defaultKey, defaultAddr := tests.GenerateKeyAndAddr(t)
+
 	setupPool := func(header *types.Header) *TxPool {
 		pool, err := newTestPool(header)
 		if err != nil {
 			t.Fatalf("cannot create txpool - err: %v\n", err)
 		}
 
-		pool.EnableDev()
-		pool.SetSigner(crypto.NewEIP155Signer(100))
+		pool.SetSigner(poolSigner)
 
 		return pool
 	}
 
-	t.Run("ErrBlockLimitExceeded", func(t *testing.T) {
+	signTx := func(transaction *types.Transaction) *types.Transaction {
+		signedTx, signErr := poolSigner.SignTx(transaction, defaultKey)
+		if signErr != nil {
+			t.Fatalf("Unable to sign transaction, %v", signErr)
+		}
+
+		return signedTx
+	}
+
+	t.Run("ErrNegativeValue", func(t *testing.T) {
 		pool := setupPool(nil)
 
-		tx := newTx(addr1, 0, 1)
-		tx.Value = big.NewInt(1)
-		tx.Gas = invalidGasLimit
+		tx := newTx(defaultAddr, 0, 1)
+		tx.Value = big.NewInt(-5)
+
+		tx = signTx(tx)
 
 		assert.ErrorIs(t,
 			pool.addTx(local, tx),
-			ErrBlockLimitExceeded,
+			ErrNegativeValue,
```

Comment:
I would place the `signTx` as argument to `addTx`.

Like so (applies to all below cases):
`pool.addTx(local, signTx(tx))`

Reply:
Inlined 👍 

652b947